### ISSUE TITLE
[Feature Request] Add Autocomplete component and introduce example of async call

### ISSUE THREAD

**Issue by ajoslin103** [dohomi/react-hook-form-mui#9]
I don't know if you can do this...

Please consider somehow integrating https://www.npmjs.com/package/react-places-autocomplete, or: https://www.npmjs.com/package/material-ui-autocomplete-google-places, 

I struggled through many different versions before finding one I could get working. (and it does work quite well once you get it going) The one I got working is a bit of a mess, I know you could do much better !!

It would be really nice for my form to have an address component whose data was gathered automatically

Thanks for your time and attention !

**Comment by dohomi** [comment]
@ajoslin103 integrating `Autocomplete` is currently on low priority I will see if I find time for it. I currently don't need it myself and I am pretty backed up with work.

**Comment by ajoslin103** [comment]
I understand!  

There aren't enough cycles in the day !!

You are doing so much for me, if I can make it work I'll contribute it

**Comment by AntalyanGriffin** [comment]
Autocomplete would be incredibly useful; I couldn't use this otherwise great project for my form because of not supporting Autocomplete which I needed :/

**Comment by bryantobing12** [comment]
async autocomplete is all I needed but I still can't find the proper way to do this. :(

**Comment by dohomi** [comment]
This feature finally landed in version `3.9.0`

**Comment by dohomi** [comment]
@bryantobing12 a simple way to solve with with the current version (example with apollo-client hook).

```ts
export function AsyncAutocomplete({ name, label, multiple, variables }: TermsSelectorProps) {
  const { loading, data, error } = useQuery(DOCUMENT_QUERY)
  if (error) {
    return (
      <Alert severity={'error'}>{error?.message}</Alert>
    )
  }
  return (
    <AutocompleteElement name={name}
                         loading={loading}
                         multiple={multiple}
                         label={label}
                         options={data?.term?.map(term => ({
                           id: term.id,
                           label: term.en
                         })) ?? []} />
  )
}

```
